pkg/search/result.go

```diff
@@ -0,0 +1,143 @@
+package search
+
+import (
+	"reflect"
+	"strings"
+	"time"
+)
+
+var RepositoryFields = []string{
+	"archived",
+	"createdAt",
+	"description",
+	"fork",
+	"fullName",
+	"private",
+	"pushedAt",
+}
+
+type RepositoriesResult struct {
+	IncompleteResults bool         `json:"incomplete_results"`
+	Items             []Repository `json:"items"`
+	Total             int          `json:"total_count"`
+}
+
+type Repository struct {
+	Archived        bool      `json:"archived"`
+	CreatedAt       time.Time `json:"created_at"`
+	DefaultBranch   string    `json:"default_branch"`
+	Description     string    `json:"description"`
+	Disabled        bool      `json:"disabled"`
+	Fork            bool      `json:"fork"`
+	ForksCount      int       `json:"forks_count"`
+	FullName        string    `json:"full_name"`
+	HasDownloads    bool      `json:"has_downloads"`
+	HasIssues       bool      `json:"has_issues"`
+	HasPages        bool      `json:"has_pages"`
+	HasProjects     bool      `json:"has_projects"`
+	HasWiki         bool      `json:"has_wiki"`
+	Homepage        string    `json:"homepage"`
+	ID              int64     `json:"id"`
+	Language        string    `json:"language"`
+	License         License   `json:"license"`
+	MasterBranch    string    `json:"master_branch"`
+	Name            string    `json:"name"`
+	OpenIssuesCount int       `json:"open_issues_count"`
+	Owner           User      `json:"owner"`
+	Private         bool      `json:"private"`
+	PushedAt        time.Time `json:"pushed_at"`
+	Size            int       `json:"size"`
+	StargazersCount int       `json:"stargazers_count"`
+	UpdatedAt       time.Time `json:"updated_at"`
+	WatchersCount   int       `json:"watchers_count"`
+
+	// URLs
```

Comment:
Listing all these `*URL` fields in the Go struct doesn't seem necessary since we don't consume them at all. I would vote that we omit them until some of them are necessary.